apps/src/javalab/backpack.module.scss

```diff
@@ -93,3 +93,30 @@
   margin-top: 10px;
   margin-bottom: 0;
 }
+
+%backpack-button {
+  color: $white;
+  font-size: 14px;
+  padding: 5px 10px;
+  width: fit-content;
+}
+
+.buttonRow {
+  display: flex;
+  justify-content: flex-end;
+  padding: 0 10px 5px 10px;
+}
+
+.button {
```

Comment:
I don't have a strong opinion here. I was following [this pattern](https://github.com/code-dot-org/code-dot-org/blob/ef0bf5945d7c29c01a7221717d97b6ad7e438a5e/apps/src/javalab/control-buttons.module.scss#L3) but if what you put is preferable I can switch! If we put the styles inside `.button` do we need the `@extend` for the inner ones? I don't know this syntax very well yet.

Reply:
yeah, unfortunately you do have to extend even if the styles are nested, which is kind of annoying. i made a little gist to show why that's true (left side is SCSS, right side is how it's compiled to CSS): https://www.sassmeister.com/gist/5bd896f327ef27dc42c51c80ffda5fcd

i don't have strong feelings, so let's leave them separate for now. it's easy enough to combine them in the future

Reply:
thanks! that tool is helpful :)

Reply:
Could we remove the extend by making the `&Orange` just a standard inner class and then using both `button` and `orange` classnames? i.e.
```
.button {
   color: $white;
   font-size: 14px;
   ...

   .orange {
      color: $orange;
      ...
   }
}
```
```
// in JS
className={classNames(styles.button, styles.orange)}
```

Reply:
yes, that definitely works! during summer swarm, people didn't really like having to apply two classes so we started using the `@extend` pattern, but i don't think that needs to be a team-wide decision since the resulting CSS is the same. in some cases, this pattern is more useful, so i don't think we need to choose between the two

Reply:
I'm going to leave this as is since it's a pattern we are already using :)